database/database.go

```diff
@@ -0,0 +1,88 @@
+package database
+
+import (
+	"dashboard/model"
+	"database/sql"
+	"log"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+var db *sql.DB
+
+//InitialiseDb to start db server
+func InitialiseDb() {
+	var err error
+	db, err = sql.Open("mysql", "root:mysql1234@tcp(localhost:3306)/dashboard")
+	if err != nil {
+		log.Panic(err.Error())
+	}
+}
+
+//CloseDb to close DB connection
+func CloseDb() {
+	db.Close()
+}
+
+//GetUserByUsername func for sign_up
+func GetUserByUsername(user *model.UserProfile) (*model.UserProfile, error) {
+	var err error
+	stmtOut, err := db.Prepare("SELECT username FROM user_profile WHERE username  = ? ")
+	if err != nil {
+		panic(err.Error())
+	}
+	defer stmtOut.Close()
+	var username string
+	err = stmtOut.QueryRow(user.Username).Scan(&username)
+	return user, err
+}
+
+//InsertUserProfile func for sign_up
+func InsertUserProfile(user *model.UserProfile) error {
+	var err error
+	_, err = GetUserByUsername(user)
+	if err != nil {

```

Comment:
It should be 
profile, err = GetUserProfileByUsername(data.username)
if profile == nil and err != nil {
// insertion code
}